batchflow/opensets/pascal.py

```diff
@@ -32,92 +31,88 @@ class BasePascal(ImagesOpenset):
-            urllib.request.urlretrieve(self.SOURCE_URL, localname)
-            logger.info("Downloaded %s", filename)
-            if self.bar:
-                self.bar.update(1)
-
-        return localname
-
-    def _name(self, file):
+            r = requests.get(self.SOURCE_URL, stream=True)
+            file_size = int(r.headers['Content-Length'])
+            chunk = 1
+            chunk_size = 1024
+            num_bars = int(file_size / chunk_size)
+            with open(localname, 'wb') as f:
+                for chunk in tqdm.tqdm(r.iter_content(chunk_size=chunk_size), total=num_bars, unit='KB',
+                                       desc=filename, leave=True):
+                    f.write(chunk)
+
+    def _name(self, path):
         """ Return file name without format """
-        return basename(file.name).split('.')[0]
+        return basename(path).split('.')[0]
+
+    def _imagepath(self, name):
+        """ Return the path to the .jpg image in the archive by its name """
+        return os.path.join(dirname(self.SETS_PATH), 'JPEGImages', name + '.jpg')
 
     def _extract(self, archive, file):
         data = archive.extractfile(file).read()
         return PIL.Image.open(BytesIO(data))
 
     def _get_ids(self, archive, part):
         """ Train and test images ids are located in specific for each task folder"""
-        part_path = os.path.join(self.sets_path, self.task, part) + '.txt'
+        part_path = os.path.join(self.SETS_PATH, self.task, part) + '.txt'
         raw_ids = archive.extractfile(part_path)
         list_ids = raw_ids.read().decode().split('\n')
-        return list_ids
-
-    def _is_train_image(self, file, train_ids):
-        return self._name(file) in train_ids \
-               and basename(dirname(file.name)) == 'JPEGImages'
-
-    def _is_test_image(self, file, test_ids):
-        return self._name(file) in test_ids \
-               and basename(dirname(file.name)) == 'JPEGImages'
-
+        return list_ids[:-1]
 
 class PascalSegmentation(BasePascal):
     """ Contains 2913 images and masks.
 
     Notes
     -----
-    - Index 0 corresponds to background and index 255 corresponds to 'void' or unlabelled.
+    Index 0 corresponds to background and index 255 corresponds to 'void' or unlabelled.
     """
     task = 'Segmentation'
-    name_classes = ['background', 'aeroplane', 'bicycle', 'bird', 'boat', 'bottle', 'bus', 'car', 'cat', 'chair',
-                    'cow', 'diningtable', 'dog', 'horse', 'motorbike', 'person', 'pottedplant', 'sheep', 'sofa',
-                    'train', 'tvmonitor', 'void']
-
-    def _is_train_mask(self, file, train_ids):
-        return self._name(file) in train_ids and basename(dirname(file.name)) == 'SegmentationClass'
+    classes = ['background', 'aeroplane', 'bicycle', 'bird', 'boat', 'bottle', 'bus', 'car', 'cat', 'chair',
+               'cow', 'diningtable', 'dog', 'horse', 'motorbike', 'person', 'pottedplant', 'sheep', 'sofa',
```

Comment:
Why did you remove the spaces (e.g. in 'dining table')? Get them back. It would be convenient to use these names in visualisation